### PR TITLE
refactor(components): BREAKING CHANGE: remove cancel from button and uses of button [JOB-37117]

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

We have removed all instances of `cancel` in uses of Atlantis button; this removes the cancel variation from Button itself 

### Removed

- cancel, replaced with subtle

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by joelwarrington-jobber** [comment]
@chris-at-jobber maybe this should be marked as a breaking change, so it triggers a major version bump

**Review by joelwarrington-jobber — CHANGES_REQUESTED** [review]
@chris-at-jobber can you merge latest master in to resolve conflicts?

Afterwards, should be okay to review, and likely approve!

**Comment by chris-at-jobber** [comment]
@joelwarrington-jobber if you have some time this is ready for a look!

File: packages/components/src/Button/Button.mdx [hunk 1]
```diff
@@ -182,10 +182,10 @@ elements such as a Modal header.
         <Button label="Cancel" type="primary" variation="subtle" />
       </Content>
       <Content>
-        <Button label="Cancel" type="secondary" variation="subtle" />
+        <Button label="Dismiss" type="secondary" variation="subtle" />
```
Comment (chris-at-jobber) [inline packages/components/src/Button/Button.mdx:185]: Showing a broader range of examples

File: packages/components/src/Button/Button.tsx [hunk 2]
```diff
@@ -80,7 +81,7 @@ interface SubmitButtonProps
 
 export type ButtonProps = XOR<
   BaseActionProps,
-  XOR<CancelActionProps, SubmitActionProps>
+  XOR<DestructiveActionProps, SubmitActionProps>
```
Comment (chris-at-jobber) [inline packages/components/src/Button/Button.tsx:84]: I think this is appropriate use of XOR?

**Review by joelwarrington-jobber — APPROVED** [review]
LGTM

`cancel` variation no longer works and is replaced by `subtle`
image
image

ConfirmationModal working as expected:
image


Have also confirmed locally that the Typing of Button is accurate to what we want.
- Error when trying to use `cancel`
- `type` restrictions on Submit type button

**Comment by adambobadam** [comment]
LGTM, just wondering why `cancel` is beeing yeeted in the first place.  I.e., are there any instances where using `subtle` instead of `cancel` wouldn't make sense? Or where `cancel` might still be needed?

**Comment by chris-at-jobber** [comment]
@adambobadam more details on the initial work [here!](https://jobber.atlassian.net/browse/JOB-34785)

**Comment by adambobadam** [comment]
> @adambobadam more details on the initial work [here!](https://jobber.atlassian.net/browse/JOB-34785)

Thanks for that! 🙏

**Comment by chris-at-jobber** [comment]
Also chatted offline w/ @adambobadam 

image

**Pull request merged**